documentation/List-of-Action-plugins.md

```diff
@@ -142,31 +143,86 @@ This will look for a user record with `username = "js"` and update it with the _
+
+An example:
+
+```
+{
+  pattern: "<?tempCode>-[A-Z]{3}-<+dddd>",
+  counterName: "demo",
+  counterInit: 100,
+  customFields: {
+    tempCode: "applicationData.templateCode"
+}
+```
+
+The pattern `<?tempCode>-[A-Z]{3}-<+dddd>` consists of:
+
+- `<?tempCode>` - this will be replaced with whatever data field is specified in `customFields` -- in this case `applicationData.templateCode`. Note that this is equivalent to just putting `<applicationData.templateCode>` directly in the pattern.
+- `[A-Z]{3}` a regular expression specifying 3 random upper-case characters in the A to Z range
+- `<+dddd>` will be replaced with a four-digit number, which will be taken from the current value of the counter specifed in `counterName`, in this case "demo".
 
-`formatExpression` can be either a string, or a full evaluator expression. If it's a string, it should use `${property.name}` for values to be replaced from `applicationData` (plus any additional fields from `additionalData`). For example, the formatExpression string:
-`${applicationData.templateName} — ${applicationData.applicationSerial}`  
-might output "Drug Registration — 12345" for a Drug Registration application. If no formatExpression is provided, it will default to this example expression.
+For an "Organisation Registration" application, this pattern would result in a sequence of output strings like:  
+`OrgRegistration-KUD-0100`  
+`OrgRegistration-JNI-0101`  
+`OrgRegistration-QLF-0102`
 
-The parameters `fieldName`, `matchField`, and `matchValue` work exactly the same as the ["ModifyRecord" Action](#modify-record) (above). These are not required if just updating the application name (that's the default behaviour)
+If the counter doesn't already exist in the `counter` table, it will be created when first called, and its value will be set to the value specified in `counterInit` (or 1 if `counterInit` not set).
 
-If `shouldUpdateDatabase` is set to `false` the string will be generated (and passed to Output) without touching the database.
+#### Updating a record
 
-`fallbackString` is an alternative value to insert for substitutions in `formatExpression` that fail (i.e. they specify an object property that doesn't exist).
+This action can also update a database record with the generated value, by specifying `updateRecord: true`
+
+The parameters `tableName`, `fieldName`, `matchField`, and `matchValue` work exactly the same as the ["ModifyRecord" Action](#modify-record) (above). These are not required if just updating the application name (that's the default behaviour).
+
+The parameters to update an application name might be something like:
+
+```
+  pattern: "<?templateName>—<?productName>",
+  customFields: {
+      templateName: "applicationData.templateName",
+      productName: "applicationData.responses.Q20.text"
+  }
+```
+
+which is the same as just:
+
+```
+  pattern: "<applicationData.templateName>—<applicationData.responses.Q20.text>"
+```
```

Comment:
I think its confusing that this example is showing how to use this action with its default option, when this topic is about using this to be "Updating a record" that isn't the applicationName. Can you move this example to the top and add another one here on how to use this to update a different record?